internal/engine/actions/actions.go

```diff
@@ -195,65 +255,70 @@ func shouldAlert(
 	}
 
 	// Proceed with use cases where the evaluation changed
-
-	// Case 3 - Evaluation changed from something else to PASSING -> Alarm should be OFF
-	if db.EvalStatusTypesSuccess == newEval {
+	switch newEval {
+	case db.EvalStatusTypesError:
+	case db.EvalStatusTypesFailure:
+		// Case 3 - Evaluation changed from something else to ERROR -> Alert should be ON
+		// Case 4 - Evaluation has changed from something else to FAILED -> Alert should be ON
+		// The Alert should be on (if it wasn't already)
+		if db.AlertStatusTypesOn != prevAlert {
+			return engif.ActionCmdOn
+		}
+		// We should do nothing if alert was already turned on
+		return engif.ActionCmdDoNothing
+	case db.EvalStatusTypesSuccess:
+		// Case 5 - Evaluation changed from something else to PASSING -> Alert should be OFF
 		// The Alert should be turned OFF (if it wasn't already)
 		if db.AlertStatusTypesOff != prevAlert {
 			return engif.ActionCmdOff
 		}
 		// We should do nothing if the Alert is already OFF
 		return engif.ActionCmdDoNothing
-	}
-
-	// Case 4 - Evaluation has changed from something else to FAILED -> Alarm should be ON
-	if db.EvalStatusTypesFailure == newEval {
-		// The Alert should be turned ON (if it wasn't already)
-		if db.AlertStatusTypesOn != prevAlert {
-			return engif.ActionCmdOn
-		}
-		// We should do nothing if the Alert is already ON
+	case db.EvalStatusTypesSkipped:
+	case db.EvalStatusTypesPending:
 		return engif.ActionCmdDoNothing
 	}
+
 	// Default to do nothing
 	return engif.ActionCmdDoNothing
```

Comment:
Is this logic the same as 191-216?

Reply:
Yes, I decided to refactor the separate `if` statements into a single switch now that we have duplicate behaviour for different evaluation statuses, i.e. `db.EvalStatusTypesError` and `db.EvalStatusTypesFailure`